src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1516,6 +1525,291 @@ def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=Non
+        if "CLUSTER_TYPE" in line:
+            yaml_lines[index] = '  CLUSTER_TYPE: connectedCluster'
+
+    deployment_yaml = '\n'.join(yaml_lines)
+
+    fd, temp_yaml_path = tempfile.mkstemp()
+    temp_yaml_file = os.fdopen(fd, 'w+t')
+    try:
+        temp_yaml_file.write(deployment_yaml)
+        temp_yaml_file.flush()
+        temp_yaml_file.close()
+        print(temp_yaml_file)
+        try:
+            kube_config = None
+            print("Cleaning up diagnostic container resources from the k8s cluster if existing")
+
+            subprocess_cmd = ["kubectl", "delete",
+                             "serviceaccount,configmap,daemonset,secret",
+                             "--all", "-n", "aks-periscope", "--ignore-not-found"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+            
+            subprocess_cmd = ["kubectl", "delete",
+            "ClusterRoleBinding",
+            "aks-periscope-role-binding", "--ignore-not-found"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+            subprocess_cmd = ["kubectl", "delete",
+                             "ClusterRole",
+                             "aks-periscope-role", "--ignore-not-found"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+        
+            subprocess_cmd = ["kubectl", "delete",
+                             "--all",
+                             "apd", "-n", "aks-periscope", "--ignore-not-found"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+            subprocess_cmd = ["kubectl", "delete",
+                             "CustomResourceDefinition",
+                             "diagnostics.aks-periscope.azure.github.com", "--ignore-not-found"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+            print()
+            print("Deploying diagnostic container on the K8s cluster")
+            subprocess_cmd = ["kubectl", "apply", "-f", temp_yaml_path, "-n", "aks-periscope"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.check_output(subprocess_cmd, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as err:
+            raise CLIError(err.output)
+    finally:
+        os.remove(temp_yaml_path)
+
+    print()
+    log_storage_account_url = f"https://{storage_account_name}.blob.core.windows.net/"
```

Comment:
This url is incomplete, we will have to construct complete url based on container name, as aks_kollect does. Will open a thread with you and safeer.

Reply:
Yes, I agree. For the MVP, the easy fix may be to comment out the code related to this URL, but we should still discuss this for the next iteration of the troubleshooting tool

Reply:
Okay, sounds good then. The only thing remaining is which storage container should i upload the log file from non-periscope checks. As of now, I will make a call to upload it on different storage container.

Later when we close down of things, both these problems will be fixed.